classes/Mail.php

```diff
@@ -182,6 +182,8 @@ function ($carry, $item) {
                 },
                 true
             );
+        } elseif (is_array($keepGoing) && empty($keepGoing)) {
```

Comment:
@PierreRambaud @Quetzacoalt91 In the previous implementation of this logic, since `$keepGoing` was assigned as the result of the `array_reduce()` operation, and `true` was given as the 3rd argument of `array_reduce()`, what was happening was that **when the Hook call would return an empty array, `$keepGoing` value was true**. Then the logic would continue.

With Jonathan modification, we compte 1st the result of the Hook call in `$keepGoing` then we perform the `array_reduce()` **only if `$keepGoing` is valid** using the infamous `if ($keepGoing)`. So we got a faulty behavior when the Hook call would return an empty array because `if ($keepGoing)` would be false, then the `array_reduce()` would not be executed and consequently `$keepGoing` would not be converted to `true`, which would stop the execution of the logic.

I see 2 ways to fix it:
1) revert the modification of Jonathan to restore previous logic
2) handle the empty array usecase as I did to force `$keepGoing` to true

Reply:
If I was sure of what are the possible values to be returned by `Hook::exec('actionEmailSendBefore',...)` I could handle it more properly. But I'm not sure.

Reply:
Can you please change by 
```php
if (empty($keepGoing)) {
    // Continue since there is no hook called
    $keepGoing = true;
} else {
    $keepGoing = array_reduce(
        $keepGoing,
        function ($carry, $item) {
            return ($item === false) ? false : $carry;
        },
        true
    );
}


if (!$keepGoing) {
    return true;
}
```
With a better comment than mine? ^^

Reply:
actually I changed this code because during install process, the hook always returned `null` which kicked a warning about using null instead of array in `array_reduce`
My modif was clearly not ideal, my bad But I'm afraid this one changes the behaviour as well
Maybe we should simply check after the hook:
```php
if (null === $keepGoing) {
    return true;
}
```

This will keep the previous behaviour and avoid the warning